### PR TITLE
Add search

### PR DESCRIPTION
**Pull request by pan-alex**
Description of the PR:
- Added search feature on the `/flix` route. Users enter a movie title into the input and hit submit, and the page displays key information about the movie.
- The search by movie title is case insensitive. It first searches for exact matches. If no exact match is found, it will search by partial match.
- The movie data comes from the `sample_mflix` sample dataset from MongoAtlas. The MongoAtlas account needs to be configured to load in these datasets before merging this PR. If reviewing this code, message me so that I can provide a DB string for an already-configured database.

Changes in this PR include:
- Add search schema to interact with the MongoDB (sample_mflix database)
- Add search routes & controller to receive user input, and respond with a json object containing movie data.
- Update views/flix.ejs and public/main.js to make the request and display the data in the DOM once loaded.
- Added very minimal CSS styling to replicate the rough layout of the mock-up. 

image

Co-authored by @edward2550

### CODE REVIEW

**Comment by pan-alex** [comment]
This is to address issue #11

File: public/js/main.js [hunk 1]
```diff
@@ -63,7 +63,40 @@ async function markIncomplete(){
...
+
+document.getElementById('search-movie').addEventListener('click', getMovie)
+
+async function getMovie() {
+    let givenTitle = document.getElementById('text-movie').value;
+    try {
+        const response = await fetch(`search/?title=${givenTitle}`);
+        const data = await response.json().then(x => x);
+        addMovieDataToDOM(data[0]);
+
```
Comment (floorrinse) [inline public/js/main.js:79]: There's currently no way for a user to know if their input is valid bc there's no check for their input. In the `try` block, the code is always fetching and responding even if the input is not valid.

Can we put in `if else` in the `try` here to check if their input can be found?

something like this:
```
try {
 const response = await fetch(`search/?title=${givenTitle}`);
 const data = await response.json().then(x => x);
  if (data.length !== 0) {
   addMovieDataToDOM(data[0]);
  }
  else {
   alert('Media not found. Please try another movie/tv show.');
  }
```

File: views/flix.ejs [hunk 2]
```diff
@@ -1,32 +1,54 @@
...
     </ul>
 
-    <h2><%= user.userName %> has <%= left %> movies & TV shows to watch.</h2>
-
-    <form action="/flix/createFlix" method='POST'>
-        <input type="text" placeholder="Enter Movie/Tv Show" name='flixItem'>
-        <input type="submit">
-    </form>
+    <input id='text-movie' type="text" placeholder="Enter Movie/Tv Show" name='flixItem'>
+    <button id='search-movie'>Submit</button>
```
Comment (floorrinse) [inline views/flix.ejs:24]: Can we update `Submit` to `Search`?

File: views/flix.ejs [hunk 3]
```diff
@@ -1,32 +1,54 @@
...
+                <p>IMDB Rating: <a id="movieIMDB"></a></p>
+                <p>Rotten Tomatoes Rating: <a id="movieTomatoes"></a></p>
+            </div>
+            <div class="synopsisContainer">
+                <h3>Synopsis</h3>
+                <p id="movieSynopsis"></p>
+            </div>
+        </div>
+    </section>
+
```
Comment (floorrinse) [inline views/flix.ejs:47]: I was thinking this could be created as its own component rather than putting it on this page, but not a blocker for now, it can be another pr

**Comment by floorrinse** [comment]
I noticed some changes you made that would overwrite what's currently in the main branch. Github also is letting you know you'll have to resolve those conflicts before you merge this pr (fyi i pushed up a profile nav menu change that would be overwritten)

File: public/js/main.js (hunk 1 above)
Comment (pan-alex) [reply public/js/main.js]: Great comment. Addressed in [5b849bf](https://github.com/CadmiumRed40/FindAFlick/pull/24/commits/5b849bf7e84c274d49be9835c5a3998d4dc8fb3a).

Please mark resolved if satisfactory.

File: views/flix.ejs (hunk 2 above)
Comment (pan-alex) [reply views/flix.ejs]: Addressed in [d91369c](https://github.com/CadmiumRed40/FindAFlick/pull/24/commits/d91369cfcd3f96f18716d5d66f27145e258641cd).

**Comment by pan-alex** [comment]
> 

From what I can tell this is just the style.css? I know that several branches have made changes to the CSS; I purposely only added 3 declarations so it should be simple to resolve (by accepting both changes) at merge.

File: views/flix.ejs (hunk 3 above)
Comment (pan-alex) [reply views/flix.ejs:47]: It can be extracted out at any point, and probably should. My intent for this PR was to include a minimal amount of HTML mark-up in order to display the data returned from the server. And at that point I figured to add some rudimentary layout so that it wouldn't burn my eyes (as much).

**Comment by floorrinse** [comment]
> > 
> 
> From what I can tell this is just the style.css? I know that several branches have made changes to the CSS; I purposely only added 3 declarations so it should be simple to resolve (by accepting both changes) at merge.

There's also the `views/ejs.file` where I added in the menu component if you check out the main branch code: https://github.com/CadmiumRed40/FindAFlick/blob/main/views/flix.ejs#L11

I can add it back in later though after this is merged

**Review by floorrinse — APPROVED** [review]
Looks good 👍

**Comment by pan-alex** [comment]
- Resolved conflicts and merged with main in [4877700](https://github.com/CadmiumRed40/FindAFlick/pull/24/commits/48777007971d87e95da56df13d2a18e54250a6ee)
- Re-configured MongoDB with @TonyHerbert22ATX.

Merging PR.
